### DISCUSSION TITLE
Ideas to build with wagmi

### DISCUSSION THREAD

**Discussion by tmm** [wevm/wagmi#455] [Ideas]
CALLING ALL IDEA GUYS!!!!

Want to build something with wagmi, but not sure what to build? **This thread is all about posting ideas for others to use!**

There's lots of opportunity to use wagmi to build everything from alternate frontends for popular crypto projects to NFT minting sites to crypto games and much more. Post, comment, and upvote about ideas below, or take an idea and start building it!

Build ideas to learn, improve your portfolio, have fun.

**Comment by tmm** [comment]
## Popular Crypto Frontend

Build an alternate frontend for [ENS](https://github.com/ensdomains/ens-app), [Uniswap](https://github.com/Uniswap/interface), or [Nouns](https://github.com/nounsDAO/nouns-monorepo). All these projects are open source (linked)! You can take the core functionality (e.g. registering ENS names, swapping, bidding on a Noun) and just build that.

**Reply by kesar** [reply]
alternate frontend for gnosis multisig ( https://gnosis-safe.io/ ) if their UI gets down many teams are gonna end up w problems :P

**Reply by cesargdm** [reply]
Working on alternate frontend for ENS! https://github.com/cesargdm/ens-app

**Comment by tmm** [comment]
## Basic Network Bridge

Bridging between L1s and L2s will be increasingly important. Build a simple bridge for withdrawing and depositing (e.g. [Optimism Bridge](https://app.optimism.io/bridge)).

**Comment by tmm** [comment]
## ENS Avatar manager

A lot of people don't know they can set their ENS Avatar to a NFT (it's not the easiest to do). Create a little app for people to set their avatars to NFTs they own. Here's some [info](https://mirror.xyz/taytems.eth/juchdvvQXjCZR4vethT7OR5E2J4vrhO29bcMyf64Rfs) on how to set it up.

**Reply by cesargdm** [reply]
Working on it! https://github.com/cesargdm/ens-app

**Reply by cesargdm** [reply]
Challenge completed! https://github.com/cesargdm/ens-app

I want to add more testing, documentation and features before making it available for mainnet, but it works :D

**Comment by shunkakinoki** [comment]
Would love to hack on the weekends! Wen discord for this👀

**Reply by tmm** [reply]
No Discord yet. Trying to keep things as async and open as possible for now.
